ts-force/src/rest/restObject.ts

```diff
@@ -229,27 +238,34 @@ export abstract class RestObject extends SObject {
    * @returns {Promise<void>}
    * @memberof RestObject
    */
-  public async update(opts?: { 
-      refresh?: boolean; 
-      sendAllFields?: boolean;
-      ifModifiedSince?: Date;
-      ifUnmodifiedSince?: Date;
-     }): Promise<this> {
+  public async update(opts?: { refresh?: boolean; sendAllFields?: boolean; headers?: RequestHeadersInput[] }): Promise<this> {
     opts = opts || {};
     if (this.id == null) {
       throw new Error('Must have Id to update!');
     }
     if (opts.refresh === true) {
       return this.updateComposite(opts.sendAllFields);
     } else {
-      const headers = {};
-      if (opts.ifModifiedSince) headers['If-Modified-Since'] = opts.ifModifiedSince.toUTCString()
-      if (opts.ifUnmodifiedSince) headers['If-Unmodified-Since'] = opts.ifUnmodifiedSince.toUTCString()
+      const headerOutput = this.setHeader(opts.headers);
       const response = await this._client.request.patch(
         `${this.attributes.url}/${this.id}`,
         this.toJson({ dmlMode: opts.sendAllFields ? 'update' : 'update_modified_only' }),
-        { headers }
+        { headers: headerOutput }
+      );
+
+      // Check if it contains Conditional Request Headers and deal with errors
+      const containsGetOrHeadHeader = Object.keys(opts.headers).some((key) =>
+        GetOrHeadRequestHeaders.includes(key as keyof RequestHeadersInput)
+      );
+      const containsPatchOrPostHeader = Object.keys(opts.headers).some((key) =>
+        PatchOrPostRequestHeaders.includes(key as keyof RequestHeadersInput)
       );
+      if (containsGetOrHeadHeader && response.status === 304) {
+        throw new Error('Not Modified');
+      } else if (containsPatchOrPostHeader && response.status === 412) {
+        throw new Error('Precondition Failed');
+      }
```

Comment:
why is this required?

Reply:
Also - you shouldn't be throwing generic errors
https://github.com/vacasaoss/ts-force/blob/master/ts-force/src/rest/errors.ts